[API] Remove unused PathHiderDocumentationModifier (#17030)
Sylius/Sylius: UPGRADE-API-2.0.md

        array $context = [],
    ): void;
```

* The `Sylius\Bundle\ApiBundle\OpenApi\Documentation\PathHiderDocumentationModifier` class and service have been removed.
  The `sylius.api.paths_to_hide` parameter used in this class has also been removed. We recommend defining endpoints 

>   as NotExposed in case of the need to hide operations.

Reviewer: ```suggestion
  with the `ApiPlatform\Metadata\NotExposed` class to hide specific operations.
```
